src/btree/bt_page.c

```diff
@@ -607,29 +609,46 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page)
             /* Unpack the on-page value cell. */
             __wt_row_leaf_value_cell(session, page, rip, NULL, &unpack);
             if (unpack.tw.prepare) {
-                if (!WT_TIME_WINDOW_HAS_STOP(&unpack.tw)) {
-                    /* Take the value from the original page cell. */
-                    WT_RET(__wt_page_cell_data_ref(session, page, &unpack, &buf));
-
-                    WT_RET(__wt_upd_alloc(session, &buf, WT_UPDATE_STANDARD, &upd, &size));
-                    upd->durable_ts = WT_TS_NONE;
-                    upd->start_ts = unpack.tw.start_ts;
-                    upd->txnid = unpack.tw.start_txn;
+                /* Take the value from the original page cell. */
+                WT_RET(__wt_page_cell_data_ref(session, page, &unpack, &buf));
+
+                WT_RET(__wt_upd_alloc(session, &buf, WT_UPDATE_STANDARD, &upd, &size));
+                upd->durable_ts = unpack.tw.durable_start_ts;
+                upd->start_ts = unpack.tw.start_ts;
+                upd->txnid = unpack.tw.start_txn;
+
+                /*
+                 * If the prepared update is of tombstone, instantiating both update and tombstone
```

Comment:
Wording: If the prepared update is a tombstone.

Reply:
The comment is updated. Check it again.

Reply:
You still say is of tombstone, my point was that it should read, "is a tombstone"

Reply:
Fixed. Thanks.